### ISSUE TITLE
for humble-devel, colcon build throws a bunch of warnings

### ISSUE THREAD

**Issue by kruusamae** [robotont/robotont_driver#19]
it still builds but i would be nice to clean up the colcon build output @Veix123 
```
Processing: robotont_driver]                             
--- stderr: robotont_driver                                    
/home/karl/robtech_ws/src/robotont_driver/src/driver.cpp: In member function ‘void robotont::Driver::initialize()’:
/home/karl/robtech_ws/src/robotont_driver/src/driver.cpp:25:10: warning: variable ‘plugin_power_supply’ set but not used [-Wunused-but-set-variable]
   25 |     bool plugin_power_supply;
      |          ^~~~~~~~~~~~~~~~~~~
/home/karl/robtech_ws/src/robotont_driver/src/driver.cpp:26:10: warning: variable ‘plugin_range’ set but not used [-Wunused-but-set-variable]
   26 |     bool plugin_range;
      |          ^~~~~~~~~~~~
/home/karl/robtech_ws/src/robotont_driver/src/driver.cpp: In member function ‘void robotont::Driver::initialize()’:
/home/karl/robtech_ws/src/robotont_driver/src/driver.cpp:25:10: warning: variable ‘plugin_power_supply’ set but not used [-Wunused-but-set-variable]
   25 |     bool plugin_power_supply;
      |          ^~~~~~~~~~~~~~~~~~~
/home/karl/robtech_ws/src/robotont_driver/src/driver.cpp:26:10: warning: variable ‘plugin_range’ set but not used [-Wunused-but-set-variable]
   26 |     bool plugin_range;
      |          ^~~~~~~~~~~~
/home/karl/robtech_ws/src/robotont_driver/src/plugin_base.cpp: In member function ‘virtual void robotont::PluginBase::packetReceived(const RobotontPacket&)’:
/home/karl/robtech_ws/src/robotont_driver/src/plugin_base.cpp:17:55: warning: unused parameter ‘packet’ [-Wunused-parameter]
   17 | void PluginBase::packetReceived(const RobotontPacket& packet)
      |                                 ~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/home/karl/robtech_ws/src/robotont_driver/src/plugin_odom.cpp: In member function ‘void robotont::PluginOdom::packetReceived(const std::vector<std::__cxx11::basic_string<char> >&)’:
/home/karl/robtech_ws/src/robotont_driver/src/plugin_odom.cpp:50:25: warning: catching polymorphic type ‘class std::exception’ by value [-Wcatch-value=]
   50 |   catch (std::exception e)
      |                         ^
/home/karl/robtech_ws/src/robotont_driver/src/plugin_base.cpp: In member function ‘virtual void robotont::PluginBase::packetReceived(const RobotontPacket&)’:
/home/karl/robtech_ws/src/robotont_driver/src/plugin_base.cpp:17:55: warning: unused parameter ‘packet’ [-Wunused-parameter]
   17 | void PluginBase::packetReceived(const RobotontPacket& packet)
      |                                 ~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/home/karl/robtech_ws/src/robotont_driver/src/plugin_odom.cpp: In member function ‘void robotont::PluginOdom::packetReceived(const std::vector<std::__cxx11::basic_string<char> >&)’:
/home/karl/robtech_ws/src/robotont_driver/src/plugin_odom.cpp:50:25: warning: catching polymorphic type ‘class std::exception’ by value [-Wcatch-value=]
   50 |   catch (std::exception e)
      |                         ^
In file included from /home/karl/robtech_ws/src/robotont_driver/src/plugin_motors.cpp:1:
/home/karl/robtech_ws/src/robotont_driver/include/robotont_driver/plugin_motors.hpp: In constructor ‘robotont::PluginMotors::PluginMotors(robotont::HardwarePtr, rclcpp::Node::SharedPtr)’:
/home/karl/robtech_ws/src/robotont_driver/include/robotont_driver/plugin_motors.hpp:31:15: warning: ‘robotont::PluginMotors::hw_ptr_’ will be initialized after [-Wreorder]
   31 |   HardwarePtr hw_ptr_;
      |               ^~~~~~~
/home/karl/robtech_ws/src/robotont_driver/include/robotont_driver/plugin_motors.hpp:29:27: warning:   ‘rclcpp::Node::SharedPtr robotont::PluginMotors::node_’ [-Wreorder]
   29 |   rclcpp::Node::SharedPtr node_;
      |                           ^~~~~
/home/karl/robtech_ws/src/robotont_driver/src/plugin_motors.cpp:8:1: warning:   when initialized here [-Wreorder]
    8 | PluginMotors::PluginMotors(HardwarePtr hw_ptr, rclcpp::Node::SharedPtr node_) : hw_ptr_(hw_ptr), node_(node_)
      | ^~~~~~~~~~~~
In file included from /home/karl/robtech_ws/src/robotont_driver/src/plugin_motors.cpp:1:
/home/karl/robtech_ws/src/robotont_driver/include/robotont_driver/plugin_motors.hpp: In constructor ‘robotont::PluginMotors::PluginMotors(robotont::HardwarePtr, rclcpp::Node::SharedPtr)’:
/home/karl/robtech_ws/src/robotont_driver/include/robotont_driver/plugin_motors.hpp:31:15: warning: ‘robotont::PluginMotors::hw_ptr_’ will be initialized after [-Wreorder]
   31 |   HardwarePtr hw_ptr_;
      |               ^~~~~~~
/home/karl/robtech_ws/src/robotont_driver/include/robotont_driver/plugin_motors.hpp:29:27: warning:   ‘rclcpp::Node::SharedPtr robotont::PluginMotors::node_’ [-Wreorder]
   29 |   rclcpp::Node::SharedPtr node_;
      |                           ^~~~~
/home/karl/robtech_ws/src/robotont_driver/src/plugin_motors.cpp:8:1: warning:   when initialized here [-Wreorder]
    8 | PluginMotors::PluginMotors(HardwarePtr hw_ptr, rclcpp::Node::SharedPtr node_) : hw_ptr_(hw_ptr), node_(node_)
      | ^~~~~~~~~~~~
In file included from /home/karl/robtech_ws/src/robotont_driver/src/plugin_led_module.cpp:1:
/home/karl/robtech_ws/src/robotont_driver/include/robotont_driver/plugin_led_module.hpp: In constructor ‘robotont::PluginLedModule::PluginLedModule(robotont::HardwarePtr, rclcpp::Node::SharedPtr)’:
/home/karl/robtech_ws/src/robotont_driver/include/robotont_driver/plugin_led_module.hpp:28:15: warning: ‘robotont::PluginLedModule::hw_ptr_’ will be initialized after [-Wreorder]
   28 |   HardwarePtr hw_ptr_;
      |               ^~~~~~~
/home/karl/robtech_ws/src/robotont_driver/include/robotont_driver/plugin_led_module.hpp:27:27: warning:   ‘rclcpp::Node::SharedPtr robotont::PluginLedModule::node_’ [-Wreorder]
   27 |   rclcpp::Node::SharedPtr node_;
      |                           ^~~~~
/home/karl/robtech_ws/src/robotont_driver/src/plugin_led_module.cpp:7:3: warning:   when initialized here [-Wreorder]
    7 |   PluginLedModule::PluginLedModule(HardwarePtr hw_ptr, rclcpp::Node::SharedPtr node_) : hw_ptr_(hw_ptr), node_(node_)
      |   ^~~~~~~~~~~~~~~
In file included from /home/karl/robtech_ws/src/robotont_driver/src/plugin_led_module.cpp:1:
/home/karl/robtech_ws/src/robotont_driver/include/robotont_driver/plugin_led_module.hpp: In constructor ‘robotont::PluginLedModule::PluginLedModule(robotont::HardwarePtr, rclcpp::Node::SharedPtr)’:
/home/karl/robtech_ws/src/robotont_driver/include/robotont_driver/plugin_led_module.hpp:28:15: warning: ‘robotont::PluginLedModule::hw_ptr_’ will be initialized after [-Wreorder]
   28 |   HardwarePtr hw_ptr_;
      |               ^~~~~~~
/home/karl/robtech_ws/src/robotont_driver/include/robotont_driver/plugin_led_module.hpp:27:27: warning:   ‘rclcpp::Node::SharedPtr robotont::PluginLedModule::node_’ [-Wreorder]
   27 |   rclcpp::Node::SharedPtr node_;
      |                           ^~~~~
/home/karl/robtech_ws/src/robotont_driver/src/plugin_led_module.cpp:7:3: warning:   when initialized here [-Wreorder]
    7 |   PluginLedModule::PluginLedModule(HardwarePtr hw_ptr, rclcpp::Node::SharedPtr node_) : hw_ptr_(hw_ptr), node_(node_)
      |   ^~~~~~~~~~~~~~~
---
Finished <<< robotont_driver [52.2s]

```

**Comment by Veix123** [comment]
Yes I agree, I hope @Zhven can take a look at these warnings and address them ASAP.

**Comment by Veix123** [comment]
Fixed in d63257b
